src/codegen/compiler.cc

```diff
@@ -2946,6 +2948,41 @@ MaybeHandle<JSFunction> Compiler::GetFunctionFromEval(
   }
   DCHECK(is_compiled_scope.is_compiled());
 
+  MaybeHandle<String> new_source = ReplayingReplaceScriptContents(isolate, source);
```

Comment:
This is a lot of lines, and by no means trivial:
1. Can we agree to move so much complexity to its own functions?
2. Can you add a comment explaining what all of this is? (I imagine when I was to rebase and had to merge conflicts here, I would not know what to do.)

Reply:
I'll add a comment, there are a lot of local variables that make refactoring this into a separate function tricky.